compiler/transf.nim

```diff
@@ -371,8 +371,8 @@ proc transformYield(c: PTransf, n: PNode): PTransNode =
   # c.transCon.forStmt.len == 3 means that there is one for loop variable
   # and thus no tuple unpacking:
   if e.typ.isNil: return result # can happen in nimsuggest for unknown reasons
-  if skipTypes(e.typ, {tyGenericInst, tyAlias, tySink}).kind == tyTuple and
-      c.transCon.forStmt.len != 3:
+  if c.transCon.forStmt.len != 3 and
+      skipTypes(e.typ, {tyGenericInst, tyAlias, tySink, tyPtr}).kind == tyTuple:
```

Comment:
Wait ... what? Why is ``tyPtr`` listed here? Typo? Should be ``tyLent``?

Reply:
`tyPtr` is listed here because `takeImplicitAddr` returns `tyPtr`. I agree it is doggy approach.
Solution is to remove this check completely, it should be redundant. Sempass already cheched that if c.transCon.forStmt.len != 3 there is a tuple to unpack.